### DISCUSSION TITLE
Property 'x' does not exist on type 'Record<"id", string>'.

### DISCUSSION THREAD

**Discussion by alvamanu** [react-hook-form/react-hook-form#11002] [General]
I'm using useFieldArray and it seems impossible to assign type to an iteration of field in fields array inside this hook. The full error here:

![Screenshot 2023-10-04 at 6 40 41 AM](https://github.com/react-hook-form/react-hook-form/assets/41272874/e7316ddd-7c06-4253-b3fb-8162345f56e1)

Others are having the same issue.

https://stackoverflow.com/questions/75102410/property-number-does-not-exist-on-type-recordid-string-in-react-js

 Please update us on how to fix this. Thanks.

**Comment by Moshyfawn** [comment]
Thanks for linking to the Stackoverflow question! I took the time [to answer the question there](https://stackoverflow.com/a/77231405/10414512), but I'll repost it here for your convenience.

> In React, the context **type** doesn't propagate down the components tree. When calling `useHookFormContext` in your child `ReleaseFields` component, there isn't a way for the hook to know the type of your form values you defined in your parent component unless you explicitly pass the `control` object down for `useHookFormContext` to derive the type from.
> 
> The solution would be to import your `InitiateHoldModel` form type down in your child `ReleaseFields` component and pass it into the `useHookFormContext` hook as the first generic type.
> 
> ```tsx
> // App.tsx
> export interface InitiateHoldModel { ... }
> 
> const defaultValues: InitiateHoldModel = { ... }
> 
> function App() {
>   useHookForm({ defaultValues })
> 
>   return <form>...</form>
> }
> // ReleaseFields.tsx
> import type { InitiateHoldModel } from "./App"
> 
> function ReleaseFields() {
>   const { registerState, control } = useHookFormContext<InitiateHoldModel>() // pass the form model type here
> }
> ```
